[Plugin migration] add section to update package dependencies (#680)
strapi/documentation: docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/migrate-back-end.md

| Action                         | Migration type                                                                                               |
| ------------------------------ | ------------------------------------------------------------------------------------------------------------ |
| Update Strapi packages                 | [Automatic](#automatic-strapi-packages-update) or [manual](#manual-strapi-packages-update)                             |
| Update content-types getters   | [Automatic](#automatic-content-types-getters-update) or [manual](#manual-content-types-getters-update) |
| Update content-types relations | [Manual](#updating-content-types-relations)                                                                    |
| Update configuration           | [Manual](#updating-plugin-configuration)                                                                              |

## Updating Strapi packages

:::strapi v3/v4 comparison
Package names in Strapi v3 are prefixed by `strapi-`.

Strapi v4 uses scoped packages.
:::

> To migrate to Strapi v4, rename all Strapi packages from `strapi-package-name` to `@strapi/package-name`. You will need to do this in the `package.json` dependencies and anywhere the package is imported.

Reviewer: ```suggestion
To migrate to Strapi v4, rename all Strapi packages from `strapi-package-name` to `@strapi/package-name`. This needs to be done in the `package.json` dependencies and anywhere the package is imported.
```

Reviewer: (to comply with our https://handbook.strapi.io/user-success-manual/12-rules-of-technical-writing, esp. usage of "you")

Strapi scoped packages can be updated [automatically](#automatic-strapi-packages-update) or [manually](#manual-strapi-packages-update).

### Automatic Strapi packages update

::: caution
Codemods modify the plugin source code. Before running a command, make sure you have initialized a git repo, the working tree is clean, you have pushed your v3 plugin, and you are on a new branch.
:::

> To update Strapi scoped packages automatically, run the following two commands in a terminal:

Reviewer: ```suggestion
To update Strapi scoped packages automatically:
```

> 1. [`update-pacakge-dependencies` codemod](https://github.com/strapi/codemods/blob/main/lib/v4/migration-helpers/update-package-dependencies.js)

Reviewer: ```suggestion
1. Use the [`update-package-dependencies` codemod](https://github.com/strapi/codemods/blob/main/lib/v4/migration-helpers/update-package-dependencies.js) by running the following command:
```

    ```sh
    npx @strapi/codemods migrate:dependencies [path-to-strapi-plugin]
    ```

> 2. [`update-strapi-scoped-imports` codemod](https://github.com/strapi/codemods/blob/main/lib/v4/transforms/update-strapi-scoped-imports.js)

Reviewer: ```suggestion
2. Use the [`update-strapi-scoped-imports` codemod](https://github.com/strapi/codemods/blob/main/lib/v4/transforms/update-strapi-scoped-imports.js) by running the following command:
```

    ```sh
    npx @strapi/codemods transform update-strapi-scoped-imports [path-to-file | folder]
    ```

### Manual Strapi packages update

> To update Strapi scoped packages manually, rename all Strapi packages (e.g. `strapi-package-name`) in the `package.json` to `@strapi/package-name`. Then find all instances where the package is imported and do the same.

Reviewer: ```suggestion
To update Strapi scoped packages manually:

1. Rename all Strapi packages (e.g. `strapi-package-name`) in `package.json` to `@strapi/package-name`
2. Repeat for all instances where the package is imported.
```
